datalad/support/annexrepo.py

```diff
@@ -1636,16 +1639,18 @@ def file_has_content(self, files, allow_quick=True, batch=False):
         # TODO: Also provide option to look for key instead of path
 
         def quick_check(filename):
-            filepath = opj(self.path, filename)
-            if islink(filepath):                    # if symlink
+            filepath = self.pathobj / filename
+            if filepath.is_symlink():                    # if symlink
                 # find abspath of node pointed to by symlink
-                target_path = realpath(filepath)  # realpath OK
+                target_path = filepath.resolve()  # realpath OK
                 # TODO: checks for being not outside of this repository
                 # Note: ben removed '.git/' from '.git/annex/objects',
                 # since it is not true for submodules, whose '.git' is a
                 # symlink and being resolved to some
                 # '.git/modules/.../annex/objects'
-                return exists(target_path) and 'annex/objects' in str(target_path)
+                # XXX str(target_path) should be target_path.as_posix()
+                # but somehow it doesn't work
+                return target_path.exists() and 'annex/objects' in str(target_path)
```

Comment:
Don't know what you mean by "doesn't work", but did you try inspecting `target_path.parts` instead?

Reply:
I cannot recall. In any case, that code want run for any v6+ repos (which I realized much later), so I don't care much anymore ;-)